meeting-notes/2017-08-31--ipld-deep-dives.md

```diff
@@ -0,0 +1,78 @@
+# IPLD Deep Dives August 31
+
+**Moderator:** @diasdavid
+
+**Notetaker:** @jon-choi
+
+**Attendees** 
+* @stebalien
+* @Kubuxu
+* @Magik6k
+* @diasdavid
+* @jon-choi
+
+
+## Agenda
+
+<!-- Ensure notetaker is present before you begin -->
+- Start recording
+- Call for additional agenda items (moderator)
+- IPLD Interface Spec (@stebalien)
+- Sharded Maps on IPLD, HAMT, CHAMT (@Kubuxu)
+- IPLD Selectors/Transforms + UnixFS binary blobs (@stebalien)
+
+<!-- After each call, it is the responsibility of the notetaker to save the last
+version of the notes in a file in ipfs/pm/meeting-notes, by opening a branch and
+submitting a PR. -->
+
+## Notes
+
+* IPLD Interface Spec
+  * intro: go-ipld interface considerations @stebalien
+    * block: cid & raw data
+    * nodes are blocks (go-ipld)
+      * would like to add a method for unmarshal into a struct
+    * dag nodes
+      * don't have consistent format?
+    * this need stems from go being statically typed
+      * go doesn't have json blobs (need to serialize structs before)
+  * discussion:
+    * context compared to js-ipld @diasdavid
+      * requirements stemmed from web dev usability for ipfs
+      * serialize / deserialize 
+    * not convinced this is the best way forward (@diasdavid accurate?)
+    * javascript does not have interfaces, which is why we have separate interface tests
+      * that said, having consistent naming convention helps 
+    * go, block level doesn't understand ipld, node level does, but ... (@stebalien to fill out)
```

Comment:
In go, we have two levels, soon to be 3:

1. Blocks: Raw data, carry CIDs, don't understand IPLD.
2. Nodes: Serialized but traversable data, carry CIDs, understand IPLD.
3. Objects (abstract nodes?): Not serialized, traversable data, don't carry CIDs with them, understand IPLD. (PROPOSED)

Basically, when working with IPLD, one will be able to "unmarshal"/"deserialize" `Node`s into abstract `Object`s for editing and then serialize them back into nodes to commit them.

Reply:
@Stebalien please update the notes directly, remember that Jon is still picking up in a lot of these terms and it would be hard for him to write exactly what we would like to have in the notes with the links to what we know. He already did a great job of capturing the whole thread of the convo, now we can just tune it up for every follower and for ourselves :)

Reply:
Ah. Sorry, I forgot I could do that.

Reply:
Updated.